src/mixtures/unigmm.jl

```diff
@@ -1,20 +1,22 @@
 # Univariate Gaussian Mixture Models
 
-struct UnivariateGMM <: UnivariateMixture{Continuous,Normal}
+struct UnivariateGMM{T<:Real} <: UnivariateMixture{Continuous,Normal}
     K::Int
-    means::Vector{Float64}
-    stds::Vector{Float64}
+    means::Vector{T}
+    stds::Vector{T}
     prior::Categorical
 
-    function UnivariateGMM(ms::Vector{Float64}, ss::Vector{Float64}, pri::Categorical)
+    function UnivariateGMM{T}(ms::Vector{T}, ss::Vector{T}, pri::Categorical) where {T<:Real}
```

Comment:
Don't think we need the type parameter in the constructor `UnivariateGMM{T}`, given that it's also in the `where` clause